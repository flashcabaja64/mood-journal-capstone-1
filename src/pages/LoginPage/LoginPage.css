.login_page h1 {
  line-height: 1.2;
  text-align: center;
  font-size: 2.5rem;

  display: block;
  position: absolute;
  width: 100%;
  padding-top: 5%;
  padding-bottom: 4%;

}

.login_page {
  margin: 0 auto;
  min-width: 100vw;
  text-align: center;
  display: inline-flex;
}