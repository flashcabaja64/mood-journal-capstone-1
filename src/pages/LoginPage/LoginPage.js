import React, { Component } from 'react';
import LoginForm from '../../components/LoginForm/LoginForm';
import './LoginPage.css'
import NavContext from '../../MoodContext/NavContext'

export default class LoginPage extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => {},
    },
  }

  static contextType = NavContext

  handleLoginSuccess = (token) => {
    console.log('testing')
    //const { location, history } = this.props
    const { history } = this.props
    // const destination = (location.state || {}).from || '/home'
    this.context.setAuthToken(token)
    //const destination = '/home';
    history.push('/home')
  }

  render() {
    return (
      <div className="login_page">
        <h1>Login</h1>
        <LoginForm
          onLoginSuccess={this.handleLoginSuccess}
        />
      </div>
    )
  }
}