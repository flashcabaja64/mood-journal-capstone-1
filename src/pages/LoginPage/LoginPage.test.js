import React from 'react';
import ReactDOM from 'react-dom';
import LoginPage from './LoginPage';
import {MemoryRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'

describe('LoginPage component testing', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
  
    ReactDOM.render(<MemoryRouter><LoginPage /></MemoryRouter>, div);
  
    ReactDOM.unmountComponentAtNode(div);
  });
  
  it('renders UI as expected', () => {
    const tree = renderer
        .create(<MemoryRouter><LoginPage /></MemoryRouter>)
        .toJSON();
        expect(tree).toMatchSnapshot();
  })

  it('renders empty given no tabs', () => {
    const wrapper = shallow(<LoginPage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
