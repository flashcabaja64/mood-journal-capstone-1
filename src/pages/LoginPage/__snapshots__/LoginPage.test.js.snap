// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UI as expected 1`] = `
<div
  className="login_page"
>
  <h1>
    Login
  </h1>
  <div
    className="main-login"
  >
    <div
      className="wrap-login"
    >
      <div
        className="form-wrap-login"
      >
        <form
          className="LoginForm"
          onSubmit={[Function]}
        >
          <div
            role="alert"
          />
          <div
            className="login-input user_name"
          >
            <label
              htmlFor="LoginForm_user_name"
            >
               Username
            </label>
            <input
              className="login-user_name"
              id="LoginForm_user_name"
              name="user_name"
              placeholder="Username"
              required={true}
            />
            <span
              className="focus-input"
            />
          </div>
          <div
            className="login-input password"
          >
            <label
              htmlFor="LoginForm_password"
            >
              Password
            </label>
            <input
              className="login-password"
              id="LoginForm_password"
              name="password"
              placeholder="Password"
              required={true}
              type="password"
            />
            <span
              className="focus-input"
            />
          </div>
          <div
            className="login-btn-container"
          >
            <button
              className="login-btn"
              type="submit"
            >
              Login
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
